ui/app/appointments/services/appointmentCommonService.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+angular.module('bahmni.appointments')
+    .service('appointmentCommonService', ['$rootScope', 'ngDialog', '$state', '$translate', 'appointmentsService',
+        'confirmBox', 'checkinPopUp', 'appService', 'messagingService',
+        function ($rootScope, ngDialog, $state, $translate, appointmentsService, confirmBox, checkinPopUp, appService, messagingService) {
+            var manageAppointmentPrivilege = Bahmni.Appointments.Constants.privilegeManageAppointments;
+            var ownAppointmentPrivilege = Bahmni.Appointments.Constants.privilegeOwnAppointments;
+
+            this.isCurrentUserHasPrivilege = function (privilege, currentUserPrivileges) {
+                return !_.isUndefined(_.find(currentUserPrivileges, function (userPrivilege) {
+                    return userPrivilege.name === privilege;
+                }));
+            };
+
+            var isOwnPrivilegedUserAllowedToPerformEdit = function (appointmentProvider, currentProviderUuId) {
```

Comment:
please pluralize "appointmentProvider" - should be named "appointmentProviders"

Reply:
Done.